sdk/src/main/java/com/horizen/account/secret/PrivateKeySecp256k1.java

```diff
@@ -47,7 +44,7 @@ public SecretSerializer serializer() {
 
     @Override
     public AddressProposition publicImage() {
-        var hashedKey = Hash.sha3(this.publicKey);
+        var hashedKey = Secp256k1.sha3(this.publicKey);
         return new AddressProposition(Arrays.copyOfRange(hashedKey, hashedKey.length - Account.ADDRESS_SIZE, hashedKey.length));
```

Comment:
let's put this code as a separate method of `byte[] Secp256k1.getAddress(byte[] publicKey)`
inside: calculate hash -> get last 20 bytes.

Another point that now we do not need to cache `this.publicKey`, instead we may cache directly the `address` field.